cloudferrylib/os/image/glance_image.py

```diff
@@ -227,8 +227,8 @@ def delete_image(self, image_id):
         self.glance_client.images.update(image_id, protected=False)
         self.glance_client.images.delete(image_id)
 
-    def get_image_by_id(self, image_id):
-        for glance_image in self.get_image_list():
+    def get_image_by_id(self, image_id, is_filtering=True):
+        for glance_image in self.get_image_list(is_filtering):
```

Comment:
Actually it returns full information if it's there:

    In [1]: from glanceclient import client
    In [2]: c = client.Client('1', endpoint='http://192.168.1.2:9292', token='a98d433089424890b0f4d365edbab3b8')
    In [5]: img = c.images.get('15f970ed-546b-4e73-8707-22cc67ecf2a9')
    In [6]: img.container_format
    Out[6]: u'bare'